### PR TITLE
fix lnd restart crash

### PR DESCRIPTION
**Pull request by islandbitcoin**
#6

### CODE REVIEW

**Comment by islandbitcoin** [comment]
Also fixed the web-check and added an LND server check.

File: Dockerfile [hunk 1]
```diff
@@ -1,12 +1,13 @@
 FROM --platform=linux/arm64/v8 lightninglabs/lightning-terminal:v0.7.0-alpha
 
-RUN apk add --no-cache yq
+RUN apk add --no-cache yq vim curl
```
Comment (chrisguida) [inline Dockerfile:3]: what's `vim` for?

File: Makefile [hunk 2]
```diff
@@ -11,6 +11,9 @@ all: verify
 verify: lightning-terminal.s9pk $(S9PK_PATH)
 	embassy-sdk verify s9pk $(S9PK_PATH)
 
+install: all lightning-terminal.s9pk
```
Comment (chrisguida) [inline Makefile:14]: the `lightning-terminal.s9pk` here is redundant. that is included in `all`

File: Dockerfile (hunk 1 above)
Comment (islandbitcoin) [reply Dockerfile:3]: so we can have access to xxd, which gives me the lnd macaroon in the entrypoint.

Comment (islandbitcoin) [reply Dockerfile:3]: > what's `vim` for?

so we can have access to xxd, which gives me the lnd macaroon in the entrypoint.

File: Makefile (hunk 2 above)
Comment (islandbitcoin) [reply Makefile:14]: fixed.

File: docker_entrypoint.sh [hunk 3]
```diff
@@ -19,6 +19,17 @@ else
 	RPC_HOST="bitcoind.embassy"
 	echo "Running on Bitcoin Core..."
 fi
+while true; do
```
Comment (chrisguida) [inline docker_entrypoint.sh:22]: this while loop could be a lot shorter. see: https://unix.stackexchange.com/a/644364

File: Dockerfile (hunk 1 above)
Comment (chrisguida) [reply Dockerfile:3]: why not just install `xxd`? it's [~300x smaller](https://pkgs.alpinelinux.org/package/edge/main/aarch64/xxd) than [vim](https://pkgs.alpinelinux.org/package/edge/main/aarch64/vim)

File: docker_entrypoint.sh [hunk 4]
```diff
@@ -49,7 +60,9 @@ faraday.bitcoin.password="$RPC_PASS"
...
 echo "Starting LiT..."
-/bin/litd --uipassword=$LITD_PASS --lit-dir=~/.lit --insecure-httplisten=lightning-terminal.embassy:8443
+/bin/litd --uipassword=$LITD_PASS --lit-dir=~/.lit --insecure-httplisten=lightning-terminal.embassy:8443 & 
+$lightning_terminal_process=$!
 
 trap _term SIGTERM
-wait -n $lightning_terminal_process
\ No newline at end of file
+
+wait $lightning_terminal_process
```
Comment (chrisguida) [inline docker_entrypoint.sh:68]:
```suggestion
wait $lightning_terminal_process

```

**Review by chrisguida — CHANGES_REQUESTED** [review]
see comments

File: docker_entrypoint.sh (hunk 3 above)
Comment (islandbitcoin) [reply docker_entrypoint.sh]: thanks! fixed.

File: docker_entrypoint.sh (hunk 4 above)
Comment (islandbitcoin) [reply docker_entrypoint.sh]: fixed.

File: Dockerfile (hunk 1 above)
Comment (islandbitcoin) [reply Dockerfile:3]: whoa, didn't know that. I'll try installing it.

Comment (islandbitcoin) [reply Dockerfile]: nice, it worked. updated PR.

Comment (islandbitcoin) [reply Dockerfile]: > why not just install `xxd`? it's [~384x smaller](https://pkgs.alpinelinux.org/package/edge/main/aarch64/xxd) than [vim](https://pkgs.alpinelinux.org/package/edge/main/aarch64/vim)

nice, it worked. updated PR.

File: docker_entrypoint.sh [hunk 5]
```diff
@@ -19,6 +19,12 @@ else
 	RPC_HOST="bitcoind.embassy"
 	echo "Running on Bitcoin Core..."
 fi
+MACAROON_HEADER="Grpc-Metadata-macaroon: $(xxd -ps -u -c 1000 /mnt/lnd/admin.macaroon)"
+until curl --silent --fail --cacert /mnt/lnd/tls.cert --header "$MACAROON_HEADER" https://lnd.embassy:8080/v1/getinfo &>/dev/null
+do
+    echo "LND Server is unreachable. Are you sure the LND service is running?" 
+    sleep 5
+done
```
Comment (chrisguida) [inline docker_entrypoint.sh:27]: 👍

**Review by chrisguida — APPROVED** [review]
ship it

**Pull request merged**